src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (!(herbivore instanceof Carnivore || herbivore.hidden === true)) {
+      herbivore.health -= 50;
+
+      if (herbivore.health === 0) {
+        const index = Animal.alive.findIndex((val) => val === herbivore);
+
+        Animal.alive.splice(index, 1);
+      }
```

Comment:
[CODE PERFORMANCE] Instead of using `findIndex` and `splice` inside a conditional, it is preferred to use the `filter` method to remove dead animals from the `Animal.alive` array. This will avoid unnecessary nesting loops which can be performance costly.